sky/utils/controller_utils.py

```diff
@@ -300,7 +304,32 @@ def get_controller_resources(
                     err=f'Expected exactly one resource, got '
                     f'{len(controller_resources)} resources: '
                     f'{controller_resources}'))
-    return list(controller_resources)[0]
+    controller_resources_to_use: resources.Resources = list(
+        controller_resources)[0]
+
+    controller_exist = (global_user_state.get_cluster_from_name(controller_name)
+                        is not None)
+    if controller_exist or controller_resources_to_use.cloud is not None:
+        return {controller_resources_to_use}
```

Comment:
Why we don't have such check in this previous function? Does that mean a custom resource will cause an existing controller fail?

Reply:
We do have such a check. Here I did a refactor...

https://github.com/skypilot-org/skypilot/blob/99d0affd1b43f437a704e1aa9280c3ce355cb93c/sky/serve/core.py#L177-L181